cmd/tealdbg/local.go

```diff
@@ -524,20 +524,38 @@ func (r *LocalRunner) RunAll() error {
 
 	failed := 0
 	start := time.Now()
-	for _, run := range r.runs {
-		r.debugger.SaveProgram(run.name, run.program, run.source, run.offsetToLine, run.states)
+	pooledApplicationBudget := uint64(0)
+	credit, _ := transactions.FeeCredit(r.txnGroup, r.proto.MinTxnFee)
+	// ignore error since fees are not important for debugging in most cases
 
+	evalParams := make([]logic.EvalParams, 0, len(r.runs))
+	for _, run := range r.runs {
+		if run.mode == modeStateful {
+			if r.proto.EnableAppCostPooling {
+				pooledApplicationBudget += uint64(r.proto.MaxAppProgramCost)
+			} else {
+				pooledApplicationBudget = uint64(r.proto.MaxAppProgramCost)
```

Comment:
Is this used at all when EnableAppCostPooling is false?

Reply:
Yes, `logic.eval` will fail the program, checked manually.